user_tools/src/spark_rapids_pytools/rapids/profiling.py

```diff
@@ -25,3 +34,226 @@ class Profiling(RapidsJarTool):
+                                                                  fail_ok=False,
+                                                                  create_dir=True)
+        self.ctxt.set_ctxt('autoTunerFilePath', input_path)
+        self.ctxt.set_ctxt('autoTunerFileName', FSUtil.get_resource_name(input_path))
+        self.logger.info('WorkerInfo successfully processed into workDir [%s]', input_path)
+
+    def _process_custom_args(self):
+        """
+        Profiling tool processes extra arguments:
+        1. the worker_info argument
+        2. the clusters
+        """
+        self._process_worker_info_arg()
+        # if the workerInfo is not set, then we need to use the gpu_cluster
+        if not self.ctxt.get_ctxt('autoTunerFilePath'):
+            self._process_offline_cluster_args()
+        else:
+            self.logger.info('Skipping building of GPU_CLUSTER because WorkerInfo is defined')
+        self._process_eventlogs_args()
+
+    def _process_offline_cluster_args(self):
+        offline_cluster_opts = self.wrapper_options.get('migrationClustersProps', {})
+        self._process_gpu_cluster_args(offline_cluster_opts)
+        self._generate_autotuner_input()
+
+    def _process_gpu_cluster_args(self, offline_cluster_opts: dict = None):
+        gpu_cluster_arg = offline_cluster_opts.get('gpuCluster')
+        if gpu_cluster_arg:
+            gpu_cluster_obj = self._create_migration_cluster('GPU', gpu_cluster_arg)
+            self.ctxt.set_ctxt('gpuClusterProxy', gpu_cluster_obj)
+        else:
+            # If we are here, we know that the workerInfoPath was not set as well.
+            # Then we should fail
+            self.logger.error('The gpuCluster argument was not set. '
+                              'Please make sure to set the arguments properly by either:\n'
+                              '  1. Setting <gpu_cluster> argument and optional set <eventlogs> if '
+                              'the path is not defined by the cluster properties ; or\n'
+                              '  2. Setting both <worker_info> and <eventlogs>')
+            raise RuntimeError('Invalid Arguments: The <gpu_cluster> and <worker_info> arguments are '
+                               'not defined. Aborting Execution.')
+
+    def _generate_autotuner_file_for_cluster(self, file_path: str, cluster_ob: ClusterBase):
+        """
+        Given file path and the cluster object, it will generate the formatted input file in yaml
+        that can be used by the autotuner to run the profiling tool.
+        :param file_path: local path whether the file should be stored
+        :param cluster_ob: the object representing the cluster proxy.
+        :return:
+        """
+        self.logger.info('Generating input file for Auto-tuner')
+        worker_hw_info = cluster_ob.get_worker_hw_info()
+        worker_info = {
+            'system': {
+                'numCores': worker_hw_info.sys_info.num_cpus,
```

Comment:
Just want to clarify, is `numCores` equivalent to `num_cpus` by our definition?

Reply:
yes, it is the same. Autotuner expects the 'numCores'